cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/JsonMenu/ClosedJsonMenu/index.tsx

```diff
@@ -51,40 +55,60 @@ const styles = (theme): StyleRules => {
   };
 };
 
-const ClosedJsonMenuView: React.FC<ICreateContext & WithStyles<typeof styles>> = ({
+const DownloadJSONButton = ({ classes, downloadDisabled, onDownloadClick }) => {
+  return (
+    <Tooltip
+      title={
+        downloadDisabled
+          ? 'Download is disabled until the required fields are filled in'
+          : 'Download Plugin JSON'
+      }
+      classes={{
+        tooltip: classes.jsonViewerTooltip,
+      }}
+    >
+      <Button disabled={downloadDisabled} onClick={onDownloadClick}>
+        <GetAppIcon />
+      </Button>
+    </Tooltip>
+  );
+};
+
+const ExpandJSONViewButton = ({ classes, expandJSONView }) => {
+  return (
+    <Tooltip
+      title="Open JSON View"
+      classes={{
+        tooltip: classes.jsonViewerTooltip,
+      }}
+    >
+      <Button onClick={expandJSONView}>
+        <CodeIcon />
+      </Button>
+    </Tooltip>
+  );
+};
+
+interface IClosedJsonMenuProps extends WithStyles<typeof styles>, ICreateContext {
+  expandJSONView: () => void;
+  onDownloadClick: () => void;
+  populateImportResults: (filename: string, fileContent: string) => void;
+  JSONStatus: JSONStatusMessage;
+  downloadDisabled: boolean;
+}
+
+const ClosedJsonMenuView: React.FC<IClosedJsonMenuProps> = ({
   classes,
-  pluginName,
-  pluginType,
-  displayName,
-  emitAlerts,
-  emitErrors,
-  configurationGroups,
-  groupToInfo,
-  groupToWidgets,
-  widgetToInfo,
-  widgetToAttributes,
-  jsonView,
-  setJsonView,
-  outputName,
+  expandJSONView,
+  onDownloadClick,
+  populateImportResults,
+  JSONStatus,
+  downloadDisabled,
 }) => {
-  const widgetJSONData = {
-    pluginName,
-    pluginType,
-    displayName,
-    emitAlerts,
-    emitErrors,
-    configurationGroups,
-    groupToInfo,
-    groupToWidgets,
-    widgetToInfo,
-    widgetToAttributes,
-    outputName,
-  };
-  const downloadDisabled = pluginName.length === 0 || pluginType.length === 0;
   return (
     <div>
       <Drawer
-        open={!jsonView}
+        open={true}
```

Comment:
Is this intended?

Reply:
I made a bit of change to improve this component's structure since this review. Now, the two components `ClosedJsonMenu` and `JsonLiveViewer` will share the `<Drawer/>` component, whereas before the review they had separate initializations of `<Drawer/>`. 

With these new changes, it will make more sense for `open` to be set to `true` since either one of these two components is always open.